sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -522,6 +604,16 @@ private void processSendWork() {
                 if (workItem.isDeliveryStateProvided()) {
                     delivery.disposition(workItem.getDeliveryState());
                 }
+
+                if (workItem.getMessage() != null) {
+                    final byte[] encodedBytes = workItem.getMessage();
+                    sentMsgSize = sender.send(encodedBytes, 0, workItem.getEncodedMessageSize());
+                } else {
+                    final ReadableBuffer encodedBuffer = workItem.getEncodedBuffer();
+                    encodedBuffer.position(0);
```

Comment:
nit: 
```suggestion
                    encodedBuffer.rewind();
```

? in case there is a mark

Reply:
Thank you!, rewind is the correct api.